### DISCUSSION TITLE
Disable antialiasing while running

### DISCUSSION THREAD

**Discussion by Unwarped** [jnsmalm/pixi3d#62] [Q&A] [answered]
Hey again.
I'm not sure if this is more of a PixiJS or Pixi3D question.
Do you know of any way I could enable/disable antialiasing like a setting while the game is running?
I've heard about having to re-create the renderer/context as it is a static webgl setting at creation. Wasn't sure if this would play nicely with Pixi3D or how to go at it.
Any ideas or suggestions? Thanks!

**Comment by jnsmalm** [comment, marked as answer]
Yes, you need to recreate the PixiJS renderer with antialias true/false. Should play nicely with Pixi3D but after the renderer have been recreated you need to setup stuff again, for example the lighting environment.

**Comment by Unwarped** [comment]
Thank you for your reply jnsmalm. I took your advice and it does in fact work once I setup lighting and camera stuff after re-creating!
Very cool indeed :)
However it does seem to be leaking memory each time. 50-100mb each time. (Never seems to get GC'd until page refresh)
Should I be destroying anything else?
Not that they should be toggling it on/off like in my tests so many times over.
Just wanted to add this so my dad can help me test (the antialias drops performance on his old pc from 30 fps to like 5)
This is what I am doing so far.
`

	app.renderer.destroy(true)

	app.renderer = null;

	app.view = null;

	app.renderer = new PIXI.Renderer({ width: w, height: h, useContextAlpha: false, resolution: 1, antialias: getRandomInt(0, 1) });

	div.appendChild(app.view);
`

Thanks again for your time and help.

**Comment by Unwarped** [comment]
`app.renderer.gl.getExtension('WEBGL_lose_context').loseContext();`
Adding this line seems to fix the memory leak! I figured it would do that automatically.
